examples/single-page-app/verify.sh

```diff
@@ -0,0 +1,306 @@
+        "Fetch repos|${EXPECTED_REPOS}|/hub/users/envoydemo/repos"
+        "Fetch followers|${EXPECTED_FOLLOWERS}|/hub/users/envoydemo/followers"
+        "Fetch following|${EXPECTED_FOLLOWING}|/hub/users/envoydemo/following"
+    )
+
+    for endpoint in "${endpoints[@]}"; do
+        IFS='|' read -r log_message expected_response path <<< "$endpoint"
+        run_log "$log_message"
+        responds_with \
+            "$expected_response" \
+            "${proxy_scheme}://localhost:${proxy_port}${path}" \
+            "${COOKIES[@]}" \
+            "${curl_args[@]}"
+    done
+
+    run_log "Log out of Myhub"
+    RESPONSE=$(_curl "${curl_args[@]}" --head "${proxy_scheme}://localhost:${proxy_port}/logout")
+    echo "$RESPONSE" | grep "HTTP/1.1 302 Found"
+    echo "$RESPONSE" | grep "location: ${proxy_scheme}://localhost:${proxy_port}/"
+    echo "$RESPONSE" | grep "set-cookie: OauthHMAC=deleted"
+    echo "$RESPONSE" | grep "set-cookie: BearerToken=deleted"
+}
+
+get_js () {
+    _curl -k "https://localhost:${PORT_PROXY}" \
+        | grep "assets/index" \
+        | grep -oP '<script type="module" crossorigin src="/assets/[^"]+"></script>' \
+        | grep -oP '/assets/[^"]+' \
+        | sed 's/\/assets\///;s/".*//'
+}
+
+
+run_log "Generate an HMAC secret"
+HMAC_SECRET=$(echo "MY_HMAC_SECRET" | mkpasswd -s)
+export HMAC_SECRET
+envsubst < secrets/hmac-secret.tmpl.yml > secrets/hmac-secret.yml
+
+run_log "Start servers"
+bring_up_example
+
+test_auth http "${PORT_DEV_PROXY}"
+
+run_log "Live reload dev app"
+sed --in-place=.bak s/Envoy\ single\ page\ app\ example/DEV\ APP/g ui/index.html
+responds_with \
+    "DEV APP" \
+    "http://localhost:${PORT_DEV_PROXY}"
+mv ui/index.html.bak ui/index.html
+
+run_log "Run yarn lint"
+docker compose run --rm ui yarn lint
+
+run_log "Build the production app"
+docker compose run --rm ui build.sh
+
+run_log "Check the created routes"
+jq '.resources[0].filter_chains[0].filters[0].typed_config.route_config.virtual_hosts[0].routes' < xds/lds.yml
+
+test_auth https "${PORT_PROXY}"
+
+current_js=$(get_js)
+
+run_log "Ensure assets are served with compression"
+responds_with_header \
+    "content-encoding: gzip" \
+    "https://localhost:${PORT_PROXY}/assets/${current_js}" \
+    -k -i -H "Accept-Encoding: gzip"
+
+run_log "Rebuild production app"
+sed --in-place=.bak s/Login\ to\ query\ APIs/LOGIN\ NOW/g ui/src/components/Home.tsx
+docker compose run --rm ui build.sh
+wait_for 5 \
+    bash -c "\
+        responds_without \
+            \"$current_js\" \
+            \"https://localhost:${PORT_PROXY}\" \
+            \"-k\""
+responds_with \
+    "Envoy single page app example" \
+    "https://localhost:${PORT_PROXY}" \
+    -k
+
+run_log "Update Envoy's configuration to use Github"
+export TOKEN_SECRET=ZZZ
+envsubst < secrets/token-secret.tmpl.yml > secrets/github-token-secret.yml
+GITHUB_PROVIDED_CLIENT_ID=XXX
+sed -i "s@cluster:\ hub@cluster:\ github@g" envoy.yml
```

Comment:
Since this comes up on the docs as well, maybe a short Python template script to generate?

Reply:
no!

this is for the user to edit with their editor - please see the related docs

Reply:
and this is the script testing what the docs say

Reply:
Yeah, I know what it says right now. It just seems being able to run some local Python script to generate secrets and build these templates files seems preferable than guiding them through multiple `sed` steps. It's probably not something I'm going to argue on that strongly, just my $0.02.

Reply:
a python script would be a total distraction, and take away the very thing being demonstrated

the point is not to just magically make these changes - the piont is the user is expected to do them - this is part tutorial - and that would be the equivalent of filling in the answers for them

Reply:
I'm not sure how populating a key-value JSON and telling a script to then template the Envoy config to drive that would be a distraction, I don't think production best practices ever involve poking around inline at files in `sed` and that `sed` is a particularly friendly tool for most users.

Sorry if this seems a bit opinionated. I'd probably be willing to compromise on something "beter than sed for users who didn't grow up poking at files with sed as an intuitive interface".

Reply:
did you read *and follow the steps* in the docs ?

i really think that would help